plugin/helpers_slash.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package plugin
+
+import (
+	"strings"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/pkg/errors"
+)
+
+// SplitCommandArgs is a helper that parses command arguments into a command, action, and parameters.
+func (p *HelpersImpl) SplitCommandArgs(args *model.CommandArgs) (string, string, []string) {
+	split := strings.Fields(args.Command)
+	command := split[0]
+	parameters := []string{}
+	action := ""
+
+	if len(split) > 1 {
+		action = split[1]
+	}
+	if len(split) > 2 {
+		parameters = split[2:]
+	}
+
+	return command, action, parameters
+}
+
+// RegisterSlashCommand is a helper used to call a provided callback function for a given trigger and passes action/parameters to the callback.
+func (p *HelpersImpl) RegisterSlashCommand(args *model.CommandArgs, trigger string, callback func(string, []string, *model.CommandArgs) (*model.CommandResponse, *model.AppError)) (*model.CommandResponse, error) {
```

Comment:
If I under the implementation correctly, won't this have to be called every time in `Plugin. ExecuteCommand(...)`?
 
It would cleaner if there was a solution where you only had to call `RegisterSlashCommand` once 
to register the callback and not worry about updating `ExecuteCommand(...)`, something like https://github.com/mattermost/mattermost-server/blob/master/plugin/api.go#L23